developer_manual/digging_deeper/reference.rst

```diff
@@ -289,6 +298,161 @@ the Smart Picker.
         })
 
+
+Use the Smart Picker in clients
+-------------------------------
+
+Clients can partially support the Smart Picker features.
+
+There are 2 types of Smart Picker providers:
+
+* The ones with a custom picker component
+* The ones that support one or multiple unified search providers
+
+As the custom picker components are web components, clients might not be able (or want) to render them.
+So we are mostly interested in the second type here: The ones using unified search providers.
+
+In Nextcloud's web UI, those providers are rendered with a
+`generic search Vue component <https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/NcRichText/NcReferencePicker/NcSearch.vue>`_
+which shows a search input, lists the search result in a dropdown menu and directly submits the URL of the selected result.
+The search is done by directly querying the Unified Search OCS API. This is described later.
+
+To implement something similar to the Smart Picker in a client, it is required to know how to:
+
+* Get the provider list
+* Use the Unified Search OCS API
+* Update the providers last usage date
+
+Get the provider list
+~~~~~~~~~~~~~~~~~~~~~
+
+The list of Smart Picker providers can be obtained via an OCS endpoint.
+Each provider object contains the list of supported unified search providers.
+
+Here is the server endpoint to list the smart picker providers:
+
+.. code-block:: bash
+
+    curl -u USER:PASSWD -H "Accept: application/json" -H "ocs-apirequest: true" \
+        "https://my.nextcloud.org/ocs/v2.php/references/providers"
+
+and an example response:
+
+.. code-block:: json
+
+    {
+      "ocs": {
+    	"meta": {
+    	  "status": "ok",
+    	  "statuscode": 200,
+    	  "message": "OK"
+    	},
+    	"data": [
+    	  {
+    	    "id": "github-issue-pr",
+    	    "title": "GitHub issues, pull requests and comments",
+    	    "icon_url": "https://my.nextcloud.org/apps/integration_github/img/app-dark.svg",
+    	    "order": 10,
+    	    "search_providers_ids": [
+    	      "github-search-issues",
+    	      "github-search-repos"
+    	    ]
+    	  },
+    	  {
+    	    "id": "openstreetmap-point",
+    	    "title": "Map location (by OpenStreetMap)",
+    	    "icon_url": "https://my.nextcloud.org/apps/integration_openstreetmap/img/app-dark.svg",
+    	    "order": 10,
+    	    "search_providers_ids": [
+    	      "openstreetmap-search-location"
+    	    ]
+    	  },
+    	  {
+    	    "id": "files",
+    	    "title": "Files",
+    	    "icon_url": "https://my.nextcloud.org/apps/files/img/folder.svg",
+    	    "order": 0
+    	  }
+    	]
+      }
+    }
+
+In this example, the "files" Smart Picker provider does not support any unified search provider
+but the "github-issue-pr" one supports 2 of them and the "openstreetmap-point" support one.
+
+Use the unified search API
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here is how to search using the Unified Search OCS API:
```

Comment:
Link to https://docs.nextcloud.com//server/stable/developer_manual/digging_deeper/search.html  ?
This way whenever the docs are adjusted there is at least a way to the original search OCS docs